src/mat/4C_mat_inelastic_defgrad_factors.cpp

```diff
@@ -26,13 +26,61 @@
 #include <Teuchos_ParameterList.hpp>
 #include <Teuchos_StandardParameterEntryValidators.hpp>
 
+#include <map>
 #include <memory>
 #include <stdexcept>
 #include <utility>
 
 FOUR_C_NAMESPACE_OPEN
 namespace
 {
+  // enum class for error types in InelasticDefgradTransvIsotropElastViscoplast, used for triggering
+  // the substepping procedures
+  enum class ErrorType
```

Comment:
One question on the error handling. In other methods, you use an `int` to encode the error. Wouldn't it be nicer and more descriptive to also make use of such an enum?

Reply:
As explained below, I use an int to signal an error in the viscoplastic law evaluations anyway, to avoid providing the ErrorType enum class to it as well.  Since the conversion from `int` to `ErrorType` is then performed anyway,  I would see this as a matter of taste whether we pass ErrorTypes or ints to and from the material methods.